frontend/app/components/work-packages/work-package-cache.service.ts

```diff
@@ -50,31 +52,46 @@ export class WorkPackageCacheService {
   }
 
   updateWorkPackage(wp: WorkPackageResource) {
-    this.updateWorkPackageList([wp]);
+    // this.updateWorkPackageList([wp]);
+    states.workPackages.put(wp.id.toString(), wp);
   }
 
   updateWorkPackageList(list: WorkPackageResource[]) {
     for (const wp of list) {
-      var cached = this.workPackageCache[wp.id];
-      if (cached && cached.dirty) {
-        this.workPackageCache[wp.id] = cached;
-      } else {
-        this.workPackageCache[wp.id] = wp;
-      }
+      // var cached = this.workPackageCache[wp.id];
+
+      // TODO RR Question: This check doesn't seem to be right!?
```

Comment:
This check was used for detecting which work packages are currently being edited but not yet saved (-> dirty). For example, if you filter the WP list while editing a work package, this check ensures that the work package under edit is not being taken away from under you.